src/modules/trainer-pov/modules/onboarding/store/onboardingActions.store.ts

```diff
@@ -0,0 +1,67 @@
+import { createProfile } from '../api/onboarding.api';
+import { ProfileInfo } from '@shared/types/ProfileInfo.type';
+import { User } from '@shared/types/User.type';
+import { StoreSlice, asyncFn } from '@store/index.store';
+import { OnboardingState } from './onboardingState.store';
+
+export interface OnboardingActions {
+  // Actions
+  updateOnboarding: (payload: Partial<OnboardingState>) => void;
+  updateProgress: (payload: Partial<OnboardingState['progressState']>) => void;
+  updateApp: (payload: Partial<OnboardingState['app']>) => void;
+  updateProfile: (payload: Partial<OnboardingState['profile']>) => void;
+
+  // Async Actions
+  createProfileAsync: (payload: ProfileInfo) => Promise<User>;
+}
+
+// TODO@store: Make snippet for store, slice, and actions
+export const onboardingSliceActions: StoreSlice<OnboardingState, OnboardingActions> = (
+  set,
+) => ({
+  // #######################################################
+  // #region Actions
+
+  updateOnboarding(payload) {
+    set((state) => {
+      return { ...state, ...payload };
+    });
+  },
+  updateProfile(payload) {
+    set((state) => {
+      return { ...state, profile: { ...state.profile, ...payload } };
+    });
+  },
+  updateApp(payload) {
+    set((state) => {
+      return { ...state, app: { ...state.app, ...payload } };
+    });
+  },
+  updateProgress(payload) {
+    set((state) => {
+      Object.entries(payload).forEach(([step, progress]) => {
+        state.progressState[step] = progress;
+      });
+
+      state.progress = 0;
+      Object.values(state.progressState).forEach((progress) => {
+        state.progress += progress;
+      });
+
+      return state;
+    });
+  },
+  // #endregion Actions
+  // #######################################################
+
+  // #######################################################
+  // #region Async Actions
+
+  async createProfileAsync(payload) {
```

Comment:
A kako ces ovo da koristis, da pozoves prvo create profile async iz komponente, pa onda nakon toga i update? ne znam dal da spojimo to direktno ovde

Reply:
https://awesomedevin.github.io/zustand-vue/en/docs/basic/async to im je iz dokumentacije maltene kao recommended, ali kako god se dogovorimo

Reply:
Pa prvo create, pa posto on vraca user-a, to bi sacuvao u authStore... Tako da, da, prvo planiram da pozoveom create profile async iz komponente, pa onda update u authStore... 

Mada da ovo povlaci neke update u onbarding store, stavio bi direktno u ovu akciju.

Reply:
Ma ok za ovaj slucaj, nego generalno pricam kako da radimo da se sync.